packages/peregrine/lib/talons/CheckoutPage/useCheckoutPage.js

```diff
@@ -0,0 +1,38 @@
+import { useCallback, useState } from 'react';
+
+import { useAppContext } from '../../context/app';
+import { useUserContext } from '../../context/user';
+import { useCartContext } from '../../context/cart';
+
+export const useCheckoutPage = () => {
+    const [, { toggleDrawer }] = useAppContext();
+    const [{ isSignedIn }] = useUserContext();
+    const [{ isEmpty }] = useCartContext();
+
+    /**
+     * Using local state to maintain these booleans. Can be
+     * moved to checkout context in the future if needed.
+     */
+    const [shippingInformationDone, setShippingInformationDone] = useState(
+        false
+    );
+    const [shippingMethodDone, setShippingMethodDone] = useState(false);
```

Comment:
In the same vein as my above comment, the checkout page itself would just query for selected shipping method, and if found, would know that the current step should open to the payment input form.

Reply:
Check my reply to the above comment.